app/src/sharedTest/java/org/oppia/android/app/topic/TopicFragmentTest.kt

```diff
@@ -202,181 +186,88 @@ class TopicFragmentTest {
-        )
-      ).perform(click())
+      clickTabAtPosition(position = 2)
       testCoroutineDispatchers.runCurrent()
-      onView(
-        atPositionOnView(
-          R.id.topic_practice_skill_list,
-          0,
-          R.id.master_skills_text_view
-        )
-      ).check(
-        matches(
-          withText(
-            containsString(
-              "Master These Skills"
-            )
-          )
-        )
+      matchStringOnListItem(
+        recyclerView = R.id.topic_practice_skill_list,
+        position = 0,
+        targetViewId = R.id.master_skills_text_view,
+        stringToMatch = "Master These Skills"
       )
     }
   }
 
   @Test
   fun testTopicFragment_clickOnReviewTab_showsReviewTabSelected() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
-      onView(
-        allOf(
-          withText(TopicTab.getTabForPosition(3).name),
-          isDescendantOfA(withId(R.id.topic_tabs_container))
-        )
-      ).perform(click())
-      onView(withId(R.id.topic_tabs_container)).check(
-        matches(
-          matchCurrentTabTitle(
-            TopicTab.getTabForPosition(3).name
-          )
-        )
-      )
+      clickTabAtPosition(position = 3)
+      matchTabTitleAtPosition(position = 3)
     }
   }
 
   @Test
   fun testTopicFragment_clickOnReviewTab_showsReviewTabWithContentMatched() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
       testCoroutineDispatchers.runCurrent()
-      onView(
-        allOf(
-          withText(TopicTab.getTabForPosition(3).name),
-          isDescendantOfA(withId(R.id.topic_tabs_container))
-        )
-      ).perform(click())
+      clickTabAtPosition(position = 3)
       testCoroutineDispatchers.runCurrent()
-      onView(atPositionOnView(R.id.revision_recycler_view, 0, R.id.subtopic_title)).check(
-        matches(
-          withText(
-            containsString(
-              "What is a Fraction?"
-            )
-          )
-        )
+      matchStringOnListItem(
+        recyclerView = R.id.revision_recycler_view,
+        position = 0,
+        targetViewId = R.id.subtopic_title,
+        stringToMatch = "What is a Fraction?"
       )
     }
   }
 
   @Test
   fun testTopicFragment_clickOnReviewTab_thenInfoTab_showsInfoTab() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
-      onView(
-        allOf(
-          withText(TopicTab.getTabForPosition(3).name),
-          isDescendantOfA(withId(R.id.topic_tabs_container))
-        )
-      ).perform(click())
-      onView(
-        allOf(
-          withText(TopicTab.getTabForPosition(0).name),
-          isDescendantOfA(withId(R.id.topic_tabs_container))
-        )
-      ).perform(click())
-      onView(withId(R.id.topic_tabs_container)).check(
-        matches(
-          matchCurrentTabTitle(TopicTab.getTabForPosition(0).name)
-        )
-      )
+      clickTabAtPosition(position = 3)
```

Comment:
One suggestion: maybe create a few top-level test constants to represent the positions for each tab, e.g.:

```kotlin
private const val INFO_TAB_POSITION = 0
...
private const val REVIEW_TAB_POSITION = 3
```

Then you can do something like ``clickTabAtPosition(REVIEW_TAB_POSITION)`` which will further help with readability. I'm not sure the position is actually needed in the context of the test in this case, so I think we can abstract it away without losing information.

Reply:
Done